### DISCUSSION TITLE
questions regarding `setup.py` code

### DISCUSSION THREAD

**Discussion by am610** [LSSTDESC/firecrown#318] [General]
the `setup.py` code has these following lines, which I want to confirm wether correct:
1. 
```
with open(pth, "r") as fp:
    start = False
    for line in fp.readlines():
        if line.strip() == "dependencies:":
            start = False
        if start:
              ...

```

The variable `start` is meant to be a flag that determines when the script should start reading the dependencies from the environment.yml file ?  However, it is initialized to False and is never set to True. Additionally, there’s a line that incorrectly sets start back to False when the script encounters the "dependencies:" line, instead of setting it to True. Is this correct ?

2. In the `_munge_req` function: 
`for sym in ["~", "=", "<", ">", ",", "!", "!"]:
`
The "!" symbol is listed twice in the list. This is not redundant ?

3. `r = line.strip()[3:].strip()`
why this first 3 letters stripping ?

**Comment by marcpaterno** [comment]
This file has not been updated (except for some formatting tweaks) in several years. We need to clean it up, or perhaps to replace it by using `pyproject.toml`. For the short term, we will just fix the `setup.py`. If anyone wants to argue for the use of `pyproject.toml`, please raise the issue on the #desc-firecrown slack channel.

**Comment by marcpaterno** [comment]
The `setup.py` file is now almost empty, because we are using the `pyproject.toml` and `setup.cfg` for the  project description.
